src/libmongoc/src/mongoc/mongoc-cluster-private.h

```diff
@@ -215,6 +215,23 @@ _mongoc_cluster_get_auth_cmd_x509 (const mongoc_uri_t *uri,
                                    bson_t *cmd /* OUT */,
                                    bson_error_t *error /* OUT */);
 
+bool
+mongoc_topology_uses_server_api (const mongoc_topology_t *topology);
```

Comment:
I'm of mixed minds with this. I like your suggestion because it keeps the relevant types together. However, the implementations actually call each other, so they need to "see" each other's prototypes. Another idea is a new header/module. I will try it and see how it goes; if there's no dependency hell that seems reasonable.

Reply:
Given declarations in header files and definitions in source files, there should be no issue with circular dependencies in header file inclusion.